test/mocks/services/ruleService.mock.ts

```diff
@@ -0,0 +1,19 @@
+import httpClientMock from './httpClient.mock';
+import { RuleService } from '../../../public/services';
+
+const ruleServiceMock = {
+  getRules: () =>
+    Promise.resolve({
+      ok: true,
+      response: {
+        hits: {
+          hits: [],
+        },
+      },
+    }),
+};
+const ruleService = new RuleService(httpClientMock);
+Object.assign(ruleService, ruleServiceMock);
```

Comment:
Why do we need to instantiate actual Rule service when we are assigning the properties of mock to it? Can we not just create the mock?

Reply:
Similarly in other places

Reply:
Because of the typescript, we are not mocking the entire service, just the parts we need, if we mock a partial service and try to cast it as Service, the typescript will complain about it as we are missing other parts of the interface.